### PR TITLE
adding verifyEmail TCKs

### PR DESCRIPTION
**Pull request by mzumbadoate**

### CODE REVIEW

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+    /** Only GET and POST should be handled
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/166">#166</a>
+     * @throws Exception
+     */
+    @Test(groups=["v100", "json", "html"])
+    public void verifyEmailDoesNotHandlePut() throws Exception {
+        put(VerifyRoute)
+                .then()
+                .assertThat().statusCode(allOf(not(200), not(500)))
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy:65]: What should the status code be? Should we test explicitly for that?

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+    /** Only GET and POST should be handled
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/166">#166</a>
+     * @throws Exception
+     */
+    @Test(groups=["v100", "json", "html"])
+    public void verifyEmailDoesNotHandleDelete() throws Exception {
+        delete(VerifyRoute)
+                .then()
+                .assertThat().statusCode(allOf(not(200), not(500)))
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy:76]: same question as above.

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+     * @throws Exception
+     */
+    @Test(groups=["v100", "json"])
+    public void verifyEmailErrorsForInvalidSptokenJson() throws Exception {
+
+        given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .queryParam("sptoken", "NOTEVENCLOSETOVALID")
+                .when()
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: `when` and `then` should be aligned with `given`

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+     * Serve a default HTML form for request type text/html
+     * @throws Exception
+     */
+    @Test(groups=["v100", "html"])
+    public void verifyEmailServesHtmlForm() throws Exception {
+
+        Response response =
+                given()
+                        .accept(ContentType.HTML)
+                        .when()
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: `when` and `then` should be aligned with `given`

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+     */
+    @Test(groups=["v100", "json"])
+    public void verifyEmailForAnyEmailWhenPostingJson() throws Exception {
+
+        given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body([
+                "email": "ANYEMAIL@ANYDOMAIN.COM"
+        ])
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: alignment is off.

`when` and `then` should be aligned with `given`

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    /** Respond with error when POSTing JSON without the email parameter
+     * @throws Exception
+     */
+    @Test(groups=["v100", "json"])
+    public void verifyEmailEmailRequiredWhenPostingJson() throws Exception {
+
+        given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .when()
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: `when` and `then` should be aligned with `given`

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+        def req = given()
+                .accept(ContentType.HTML)
+                .contentType(ContentType.URLENC)
+                .formParam("email", "ANYEMAIL@ANYDOMAIN.COM")
+
+        setCSRFAndCookies(req, ContentType.HTML)
+
+        req.when()
+                .post(VerifyRoute)
+                .then()
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: `when` and `then` should be aligned with `given`

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+        def req = given()
+                .accept(ContentType.HTML)
+                .contentType(ContentType.URLENC)
+
+        setCSRFAndCookies(req, ContentType.HTML)
+
+        Response response = req.when()
+                .post(VerifyRoute)
+                .then()
```
Comment (dogeared) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy]: `when` and `then` should be aligned with `given`

**Comment by edjiang** [comment]
Hey! I'm really confused by this PR. We don't have any way to trigger custom configurations in the TCK just yet, and email verification is not enabled by the default configuration, so these tests will fail by default. 

But even if it were, it doesn't look like you're getting email verification tokens from the API, so you're not testing valid `sptoken` submission.

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy (hunk 1 above)
Comment (mzumbadoate) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy:65]: I'm not sure about this
I'm just doing the same test as the other endpoints

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy (hunk 2 above)
Comment (mzumbadoate) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy:76]: I'm just doing the same test as the other endpoints

File: src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy (hunk 1 above)
Comment (mzumbadoate) [inline src/test/groovy/com/stormpath/tck/verify/VerifyEmailIT.groovy:65]: I guess we could check for 
HTTP/1.1 405 Method Not Allowed

**Comment by nbarbettini** [comment]
Agreed, we can't use this until we have a way to switch server configurations for the TCK. @mzumbadoate I'll close this for now to clean up our outstanding PRs, but we might be able to circle back and use the code when we do have that capability.

**Pull request closed without merging**